File: habit-tracker/src/main/java/com/habit/habit_tracker/dto/auth/RegisterRequest.java
package com.habit.habit_tracker.dto.auth;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public class RegisterRequest {
    @NotBlank(message = "Username is required")
    @Size(max = 20)
    private String username;

    @NotBlank(message = "Password is required")
    @Size(max = 20)
    private String password;

    public RegisterRequest() {

    }

    public RegisterRequest(String username, String password) {
        this.username = username;
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}

File: habit-tracker/src/main/kotlin/com/habit/habit_tracker/dto/auth/RegisterRequest.kt
package com.habit.habit_tracker.dto.auth

import jakarta.validation.constraints.Size
import jakarta.validation.constraints.NotBlank

data class RegisterRequest(
    @field:NotBlank(message = "Username is required")
    @field:Size(max = 20, message = "Username must be at most 20 characters")
    val username: String,

    @field:NotBlank(message = "Password is required")
    @field:Size(min = 6, max = 20, message = "Password must be between 6 and 20 characters")
    val password: String
)